sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/TestProcessor.cs

```diff
@@ -270,26 +273,14 @@ private void EndTestRun(TestRunCompleteEventArgs e)
             }
             _cloudRunErrorParser.DisplayMessages();
         }
-        private static string GetCloudFilePath(string uri, string fileRelativePath)
+        private  void UploadBuffer(string uri, string buffer, string fileRelativePath)
         {
-            string[] parts = uri.Split(new string[] { ReporterConstants.s_sASUriSeparator }, StringSplitOptions.None);
-            string containerUri = parts[0];
-            string sasToken = parts.Length > 1 ? parts[1] : string.Empty;
-
-            return $"{containerUri}/{fileRelativePath}?{sasToken}";
-        }
-        private void UploadBuffer(string uri, string buffer, string fileRelativePath)
-        {
-            string cloudFilePath = GetCloudFilePath(uri, fileRelativePath);
-            BlobClient blobClient = new(new Uri(cloudFilePath));
-            byte[] bufferBytes = Encoding.UTF8.GetBytes(buffer);
-            blobClient.Upload(new BinaryData(bufferBytes), overwrite: true);
-            _logger.Info($"Uploaded buffer to {fileRelativePath}");
+             _blobService.UploadBufferAsync(uri, buffer, fileRelativePath);
         }
 
         private void UploadBlobFile(string uri, string fileRelativePath, string filePath)
         {
-            string cloudFilePath = GetCloudFilePath(uri, fileRelativePath);
+            string cloudFilePath = _blobService.GetCloudFilePath(uri, fileRelativePath);
             BlobClient blobClient = new(new Uri(cloudFilePath));
             // Upload filePath to Blob
             blobClient.Upload(filePath, overwrite: true);
```

Comment:
We need to move this as well to DI either in this PR or next.

Reply:
updated